### PR TITLE
Fixes Issue #141 part 1: Add more instructions in scotch demo nb

### PR DESCRIPTION
**Pull request by aluu317**
(c) Copyright IBM Corp. 2015

### CODE REVIEW

**Comment by parente** [comment]
Good start @aluu317. Definitely helps to see more descriptions in the notebook. I'd like to see it become more of a step by step tutorial of the construction of the dashboard, with a summary at the start of what it's trying to do. (e.g., "In this notebook, we're going to create a dashboard that recommends scotches based on their taste profiles. To do this, we'll use ipywidgets and jupyter_dashboards ....") It'd be great to have it be the first tutorial folks who are already familiar with ipywidgets follow to get acquainted with the dashboard feature before learning about the declarative widgets too.

As a more concrete example of what I mean by tutorial, check out what we did in https://github.com/jupyter-incubator/dashboards/blob/master/etc/notebooks/stream_demo/meetup-streaming.ipynb

* There's a description of where the notebook is headed at the top
* There's a table of contents to key sections
* There's pretty lengthy text in each section describing what "we'll" do to develop the components of the dashboard, not just descriptions of the code
* There's instructions at the end of how to lay out the notebook as a dashboard

**Comment by aluu317** [comment]
I added a couple more comments, and styling to look like the meetup streaming demo/tutorial. Let me know what you think!

**Comment by parente** [comment]
Looking good on the face of it. I'll give it a deeper read this evening and try running it to make sure there's no surprises.

File: etc/notebooks/got_scotch_demo/scotch_dashboard.ipynb [hunk 1]
```diff
@@ -354,9 +366,11 @@
     "urth_section": {}
    },
    "source": [
-    "## Radar Widget\n",
+    "### Radar Widget\n",
     "\n",
-    "Make the radar chart an ipywidget so that it updates when the selected factors change."
+    "We will call a scotch a \"factor\" for this radar visualization.\n",
```
Comment (parente) [inline etc/notebooks/got_scotch_demo/scotch_dashboard.ipynb]: We will call taste features like smokey, winey, honey, etc. "factors"  in this radar visualization.

File: etc/notebooks/got_scotch_demo/scotch_dashboard.ipynb [hunk 2]
```diff
@@ -354,9 +366,11 @@
...
+    "\n",
+    "Now we want the radar chart as an ipywidget that updates when the selected factors change, meaning when user selects a new scotch, or when they click on a similar recommended scotch to compare to the original selection. We then draw the taste profiles of these scotches into a radar chart by calling the _radar( )_ function defined above."
```
Comment (parente) [inline etc/notebooks/got_scotch_demo/scotch_dashboard.ipynb]: "when the selected scotch changes"

File: etc/notebooks/got_scotch_demo/scotch_dashboard.ipynb [hunk 3]
```diff
@@ -388,6 +402,22 @@
...
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "### Radar View\n",
+    "Next, we need to define a RadarView (an inheritance of DOMWidgetView) to watch over the similarities/recommendations table.\n",
```
Comment (parente) [inline etc/notebooks/got_scotch_demo/scotch_dashboard.ipynb]: Next, we need to define a `RadarView` that serves as the front-end view for the Python widget class we defined above. This frontend will contain the Javascript that watches for events in the similarities/recommendations table and redraws the radar plot accordingly.

File: etc/notebooks/got_scotch_demo/scotch_dashboard.ipynb [hunk 4]
```diff
@@ -429,7 +459,11 @@
     "urth_section": {}
    },
    "source": [
-    "## Dashboard Setup"
+    "## Dashboard Setup <span style=\"float: right; font-size: 0.5em\"><a href=\"#Got-Scotch?\">Top</a></span>\n",
+    "\n",
+    "Finally, we will set up the dashboard to show a short introduction line which scotch you're examining, then a scotch dropdown picker, a recommendation table and a radar chart to show how similar it is to your selection.\n",
```
Comment (parente) [inline etc/notebooks/got_scotch_demo/scotch_dashboard.ipynb]: "... a short introduction line about which ..."

"... a recommendation table, and a ..."

File: etc/notebooks/got_scotch_demo/scotch_dashboard.ipynb [hunk 5]
```diff
@@ -555,6 +628,21 @@
...
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "Now we can render the previously assigned __tmpl__ with an argument _name_ as our first scotch 'Aberfeldy'.\n",
```
Comment (parente) [inline etc/notebooks/got_scotch_demo/scotch_dashboard.ipynb]: This section is a bit confusing. Probably want to have the first paragraph as markdown above the `prompt_w` cell and the second sentence as a new markdown cell directly above the `picker_w` cell.

For the first paragraph, be sure to mention that the first scotch is just the default value and that the template will re-render whenever the user picks a new scotch from the drop down.

File: etc/notebooks/got_scotch_demo/scotch_dashboard.ipynb [hunk 6]
```diff
@@ -634,15 +735,41 @@
   {
    "cell_type": "markdown",
    "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
```
Comment (parente) [inline etc/notebooks/got_scotch_demo/scotch_dashboard.ipynb]: Move the "Powered by data ..." cell above this section. Add markdown above it saying "Finally, we'll include a footer in our dashboard with some attribution information."

**Comment by parente** [comment]
Much improved over the original. Good work.

A couple minor comments above on the diff. One final reminder to clear all the cell outputs before pushing it in the PR again. (I think you did it last time, just a friendly reminder.)

File: etc/notebooks/got_scotch_demo/scotch_dashboard.ipynb (hunk 1 above)
Comment (aluu317) [inline etc/notebooks/got_scotch_demo/scotch_dashboard.ipynb]: @parente When studying the demo, I thought "factors" were the taste categories like smokey, honey, winey, etc. so I had a couple console.log to print out the "factors", and they were actually scotch names. Can you confirm the logic?

Comment (parente) [inline etc/notebooks/got_scotch_demo/scotch_dashboard.ipynb]: Ah, yes. I remember the weird naming now. The "select.factors" event is the name of a scotch which is a key for looking up the features of that scotch (the real factors). The reasoning was that it kept the widget agnostic as to the data domain vs making it a "ScotchRadarWidget". But it's still pretty terrible and damn confusing.

How about changing the variable and event to `factors_key` or `factors_about` or something?

Comment (aluu317) [inline etc/notebooks/got_scotch_demo/scotch_dashboard.ipynb]: I'll go with factors_key

**Comment by aluu317** [comment]
Add fixes for the minor comments above

**Comment by parente** [comment]
Thanks. Looks great. Merging!

**Pull request merged**